# IMPORTING NECESSARY LIBRARIES FOR EDA

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
import matplotlib.pyplot as plt
import seaborn as sns

# READING THE DATASET USING PANDAS

In [2]:
df = pd.read_csv(r"C:\Users\LENOVO\OneDrive\Desktop\datasets\news.csv", sep= '\t')

In [3]:
df

category filename                              title  \
0     business  001.txt  Ad sales boost Time Warner profit   
1     business  002.txt   Dollar gains on Greenspan speech   
2     business  003.txt  Yukos unit buyer faces loan claim   
3     business  004.txt  High fuel prices hit BA's profits   
4     business  005.txt  Pernod takeover talk lifts Domecq   
...        ...      ...                                ...   
2220      tech  397.txt   BT program to beat dialler scams   
2221      tech  398.txt    Spam e-mails tempt net shoppers   
2222      tech  399.txt            Be careful how you code   
2223      tech  400.txt    US cyber security chief resigns   
2224      tech  401.txt   Losing yourself in online gaming   

                                                content  
0      Quarterly profits at US media giant TimeWarne...  
1      The dollar has hit its highest level against ...  
2      The owners of embattled Russian oil giant Yuk...  
3      British Airways has blamed high fuel prices f...  
4      Shares in UK drinks and food firm Allied Dome...  
...                                                 ...  
2220   BT is introducing two initiatives to help bea...  
2221   Computer users across the world continue to i...  
2222   A new European directive could put software w...  
2223   The man making sure US computer networks are ...  
2224   Online role playing games are time-consuming,...  

[2225 rows x 4 columns]

# HERE WE CAN SEE THAT WE HAVE 2225 ROWS AND 4 COLUMNS IN OUR DATASET

# COUNTING THE VALUES IN EACH CATEGORY

In [29]:
a=df["category"].value_counts()
a

sport            511
business         510
politics         417
tech             401
entertainment    386
Name: category, dtype: int64

In [66]:
figure = px.bar(df, x='category',color='category',
            title="Countries with Highest Covid Cases")
figure.show()


# WE FOUND OUT THAT  MOST OF THE  NEWS ARE OF SPORTS AND BUSINESS SIDE

# CHECKING THE NULL VALUES IN OUR DATASET

In [5]:
df.isnull().sum()

category    0
filename    0
title       0
content     0
dtype: int64

# SO THIER ARE NO NULL VALUES IN OUR DATASET SO WE CAN SKIP THE DROPPING THE NULL VALUES AND RESETTING OF  INDEX

# HERE WE WILL CHECK THE DATA TYPE OF OUR DATA 

In [8]:
df.dtypes 

category    object
filename    object
title       object
content     object
dtype: object

# WE FOUND OUT THAT ALL THE DATA IN OUR DATASET IS OF OBJECT TYPE

# NOW WE WILL ADD THE COLUMNS IN X AND Y VARIABLE FOR ML

In [10]:
x=np.array(df['title'])
y=np.array(df['category'])


# The CountVectorizer provides a simple way to both tokenize a collection of text documents and build a vocabulary of known words, but also to encode new documents using that vocabulary

In [11]:
from sklearn.feature_extraction.text import CountVectorizer

In [12]:
a=CountVectorizer()

# NOW WE WILL FIT AND TRANSFORM THE DATA FOR ML OPERATION

In [13]:
ax=a.fit_transform(x)

In [14]:
from sklearn.model_selection import train_test_split


# SPLITTING THE DATA IN FOR TESTING AND TRAINING 

In [15]:
xTrain, xTest, yTrain, yTest = train_test_split(ax, y, test_size=0.1, random_state=2)

## FOR  HEIGHER ACCURACY WE TOOK 10%DATA FOR TEST AND REST IS TRAINED

In [16]:
from sklearn.naive_bayes import MultinomialNB

# Naive Bayes classifier for multinomial model,The multinomial Naive Bayes classifier is suitable for classification with discrete features (e.g., word counts for text classification).

In [17]:
b=MultinomialNB()

# FITTING THE DATA INTO MULTINOMIALNB

In [18]:
b.fit(xTrain,yTrain)

MultinomialNB()

## ADDING A TEXTFIELD TIO CHECK THE WORK OR SENTENCE AND ITS NEWS TYPE


In [19]:
z=input("enter the text:  ")

enter the text:  TRUMP


# TRANSFORMING THE TEXT INTO ARRAY FOR PREDICTION

In [20]:
msg=a.transform([z]).toarray()

In [21]:
pre=b.predict(msg)

# PRINTING THE NEWS TYPE(YOU CAN ALSO CHECK IT)

In [22]:
print(pre)

['business']


In [23]:
from sklearn.metrics import accuracy_score

# TESTING THE ACCURACY!

In [24]:
accuracy_score(yTest,b.predict(xTest))

0.8609865470852018

# BY TESTING THE DATA SEVERAL TIMES WE HAVE CONCLUDED THAT OUR DATASET IS ACCURATE AS PER THE STANDARDS WE ACHIEVE A 86% ACCURACY WHICH IS VERY GOOD .